x/auctions/keeper/hooks.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/evmos/evmos/v18/utils"
+	"github.com/evmos/evmos/v18/x/auctions/types"
+	epochstypes "github.com/evmos/evmos/v18/x/epochs/types"
+)
+
+// BeforeEpochStart starts a new auction at the beginning of the epoch
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, _ string, _ int64) {
+	// no-op
+}
+
+// AfterEpochEnd ends the current auction and distributes the rewards to the winner
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64) {
+	// If it's not the weekly epoch, no-op
+	if epochIdentifier != epochstypes.WeekEpochID {
+		return
+	}
+
+	params := k.GetParams(ctx)
+	if !params.EnableAuction {
+		return
+	}
+
+	lastBid := k.GetHighestBid(ctx)
+	lastBidAmount := lastBid.Amount.Amount
+
+	// Distribute the awards from the last auction
+	if lastBidAmount.IsPositive() && lastBid.Sender != "" {
```

Comment:
What happens if bid amount is positive and sender is == ""? I know it should not happens but in that case the bid is not cancelled. I believe we should handle it. It is drastic but I believe we should panic.